src/server/pfs/cmds/cmds.go

```diff
@@ -1017,6 +1021,59 @@ Any pachctl command that can take a commit, can take a branch name instead.`,
 	shell.RegisterCompletionFunc(deleteBranch, shell.BranchCompletion)
 	commands = append(commands, cmdutil.CreateAliases(deleteBranch, "delete branch", branches))
 
+	searchForFileInBranch := &cobra.Command{
+		Use:   "{{alias}} <repo>@<branch-or-commit>:<path/in/pfs> [flags]",
+		Short: "search for commits with reference to <filePath> within a branch starting from <repo@commitID>",
+		Long:  "search for commits with reference to <filePath> within a branch starting from <repo@commitID>",
+		Run: cmdutil.RunFixedArgs(1, func(args []string) error {
+			file, err := cmdutil.ParseFile(project, args[0])
+			if err != nil {
+				return err
+			}
+			commit := file.Commit
+			c, err := client.NewOnUserMachine("user")
+			if err != nil {
+				return err
+			}
+			defer c.Close()
+			resp, err := c.SearchForFileInBranch(&pfs.SearchForFileInBranchRequest{
```

Comment:
If you are using are client, you need to either get out the PFS gRPC and call it with a context that you configure with the timeout.  Or you can call the method on our client which clones it and sets the context.